src/Utilities/Compiler/SymbolEqualityComparer.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis
+{
+#if !CODEANALYSIS_V3_OR_BETTER
+    internal class SymbolEqualityComparer
+    {
+        private SymbolEqualityComparer()
+        {
+
+        }
+
+#pragma warning disable CA1822 // Mark members as static
+        public bool Equals(ISymbol? symbol1, ISymbol? symbol2) =>
+#pragma warning restore CA1822 // Mark members as static
+            object.Equals(symbol1, symbol2);
```

Comment:
To be more matching with roslyn implementation

```suggestion
        public bool Equals(ISymbol? x, ISymbol? y)
        {
            if (x is null)
            {
                return y is null;
            }

            return x.Equals(y);
        }

        public int GetHashCode(ISymbol? obj)
        {
            return obj?.GetHashCode() ?? 0;
        }
```

Reply:
Thanks!